examples/fluids/src/setuplibceed.c

```diff
@@ -47,6 +47,21 @@ static PetscErrorCode CreateKSPMassOperator_Unstabilized(User user, CeedOperator
   PetscCallCeed(ceed, CeedOperatorSetField(*op_mass, "qdata", elem_restr_qd_i, CEED_BASIS_NONE, q_data));
   PetscCallCeed(ceed, CeedOperatorSetField(*op_mass, "v", elem_restr_q, basis_q, CEED_VECTOR_ACTIVE));
 
+  {  // Restore restriction and basis from the RHS function
+    CeedOperator     *sub_ops;
+    CeedOperatorField field;
+    PetscInt          sub_op_index = 0;  // will be 0 for the volume op
+
+    PetscCallCeed(ceed, CeedCompositeOperatorGetSubList(user->op_rhs_ctx->op, &sub_ops));
+    PetscCallCeed(ceed, CeedOperatorGetFieldByName(sub_ops[sub_op_index], "q", &field));
+    PetscCallCeed(ceed, CeedOperatorFieldRestoreElemRestriction(field, &elem_restr_q));
+    PetscCallCeed(ceed, CeedOperatorFieldRestoreBasis(field, &basis_q));
+
+    PetscCallCeed(ceed, CeedOperatorGetFieldByName(sub_ops[sub_op_index], "qdata", &field));
+    PetscCallCeed(ceed, CeedOperatorFieldRestoreElemRestriction(field, &elem_restr_qd_i));
+    PetscCallCeed(ceed, CeedOperatorFieldRestoreVector(field, &q_data));
+  }
```

Comment:
Maybe I'm being shortsighted, but is there reason to not just require that the given restrictions, basis, and vec just be destroyed rather than restored?

In most of the backend cases it's not a problem, but seeing this code here, it does make me wonder if just forcing the user to destroy the resulting objects would be cleaner.

Reply:
I think we'd need to call it something other than Get in that case since it isn't the Get/Restore pattern.

I think Get/Restore makes the Rust interface work easier, but I haven't finished exploring that yet

Reply:
In the `let y = x.get_y()` pattern, it's easier if `Drop` for `y` does not need a reference to `x`, though we have achieved that when necessary using a closure.

Reply:
We can have the restore not require the field argument (ie. have it just be a wrapper around destroy).  That'd keep the Get/Restore pattern without having to include the `field` parameter. But that also feels like a lot of needless/confusing indirection just to match a function naming pattern.

Reply:
To clarify, I agree that keeping a Get/Restore naming pattern is useful and important. I'm maybe questioning if the function just needs a different name.

Reply:
Maybe `CeedOperatorFieldGetReference*`/`CeedOperatorFieldGetRef*``? I think that distinguishes it enough from normal Get/Restore patterns.

Reply:
Yeah, something like that is probably the way to go. I'll rebuild the branch along  those lines once the two PRs preceding this one merge